### PR TITLE
Add MQTT fan direction feature

### PR DESCRIPTION
**Pull request by rubenbe**
## Proposed change

Add MQTT fan direction documentation

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/91700
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/fan.mqtt.markdown [hunk 1]
```diff
@@ -160,6 +160,22 @@ optimistic:
...
+  required: false
+  type: string
+direction_state_topic:
+  description: The MQTT topic subscribed to receive direction state updates.
+  required: false
+  type: string
+direction_value_template:
+  description: "Defines a [template](/docs/configuration/templating/#using-templates-with-the-mqtt-integration) to extract a value from the direction."
+  required: false
+  type: string
```
Comment (jbouwh) [inline source/_integrations/fan.mqtt.markdown]:
```suggestion
  type: template
```

**Review by jbouwh — CHANGES_REQUESTED** [review]

File: source/_integrations/fan.mqtt.markdown [hunk 2]
```diff
@@ -343,6 +362,8 @@ mqtt:
     - name: "Bedroom Fan"
       command_topic: "bedroom_fan/on/set"
       command_template: "{ state: '{{ value }}'}"
+      direction_command_template '{{ iif(value == 'forward', 'fwd', 'rev') }}'
+      direction_value_template -> '{{ iif(value == 'fwd', 'forward', 'reverse') }}'
```
Comment (jbouwh) [inline source/_integrations/fan.mqtt.markdown]: Invalid syntax

Comment (jbouwh) [inline source/_integrations/fan.mqtt.markdown]:
```suggestion
      direction_command_template: "{{ iif(value == 'forward', 'fwd', 'rev') }}"
      direction_value_template: "{{ iif(value == 'fwd', 'forward', 'reverse') }}"```

**Review by jbouwh — DISMISSED** [review]

**Comment by rubenbe** [comment]
@jbouwh I noticed the templates for "oscillation_value_template", "percentage_value_template", "preset_mode_value_template" and "state_value_template" are also defined as string types instead of templates. Do I address these too here, or better in a separate PR?

**Comment by jbouwh** [comment]
> @jbouwh I noticed the templates for "oscillation_value_template", "percentage_value_template", "preset_mode_value_template" and "state_value_template" are also defined as string types instead of templates. Do I address these too here, or better in a separate PR?

Yes, we can address them in a separate PR targetted against the `current` branch. We should check all `mqtt` template config options.

File: source/_integrations/fan.mqtt.markdown [hunk 3]
```diff
@@ -343,6 +362,8 @@ mqtt:
     - name: "Bedroom Fan"
       command_topic: "bedroom_fan/on/set"
       command_template: "{ state: '{{ value }}'}"
+      direction_command_template: "{{ iif(value == 'forward', 'fwd', 'rev') }}"
+      direction_value_template: "{{ iif(value == 'fwd', 'forward', 'reverse') }}"```
```
Comment (jbouwh) [inline source/_integrations/fan.mqtt.markdown]:
```suggestion
      direction_value_template: "{{ iif(value == 'fwd', 'forward', 'reverse') }}"
```

typo (sorry)

**Review by frenck — APPROVED** [review]
Thanks, @rubenbe 👍 

../Frenck

**Pull request merged**